### PR TITLE
[Feature] Parrots with mlu adaptation, ops: nms, focal_loss_sigmoid and roi_align

### PR DESCRIPTION
**Pull request by yewentao256**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

- Adaptation for ops of mlu device in parrots
- Create a new device registry for dispatching ops

## Modification

- MLU ops
- Part of the unit test

## Use cases (Optional)

```py
import numpy as np
from mmcv.ops import nms
import torch
device = 'camb'
# device = 'cpu'

if __name__ == '__main__':
    np_boxes = np.array([[6.0, 3.0, 8.0, 7.0], [3.0, 6.0, 9.0, 11.0],
                            [3.0, 7.0, 10.0, 12.0], [1.0, 4.0, 13.0, 7.0]],
                        dtype=np.float32)
    np_scores = np.array([0.6, 0.9, 0.7, 0.2], dtype=np.float32)
    np_inds = np.array([1, 0, 3])
    np_dets = np.array([[3.0, 6.0, 9.0, 11.0, 0.9],
                        [6.0, 3.0, 8.0, 7.0, 0.6],
                        [1.0, 4.0, 13.0, 7.0, 0.2]])
    boxes = torch.from_numpy(np_boxes)
    scores = torch.from_numpy(np_scores)
    dets, inds = nms(
       boxes.to(device), scores.to(device), iou_threshold=0.3, offset=0)
    print('final_result:', dets, inds)
    assert np.allclose(dets.cpu().numpy(), np_dets)  # test gpu
    assert np.allclose(inds.cpu().numpy(), np_inds)  # test gpu
```

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials. (No docstrings here)

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by st11279 — APPROVED** [review]

File: mmcv/utils/version_utils.py [hunk 1]
```diff
@@ -19,7 +19,9 @@ def digit_version(version_str: str, length: int = 4):
     Returns:
         tuple[int]: The version info in digits (integers).
     """
-    assert 'parrots' not in version_str
+    if version_str == 'parrots':
+        return digit_version(
+            version_str.version.split('+')[0][:-2])  # type: ignore
```
Comment (chenyuwang814) [inline mmcv/utils/version_utils.py:24]: 这里我更建议给PatVersion加上一个digit方法，可以直接取到版本号，然后这里直接调用version_str.digit()

Comment (gfyts0123) [reply mmcv/utils/version_utils.py:24]: 已修改PatVersion：https://github.com/ParrotsDL/senseparrots/pull/551，该pr合入后修改此项

File: mmcv/ops/csrc/parrots/mlu/roi_align_mlu.cpp [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+  k_dim.x = getDeviceAttr(cnrtAttrMcorePerCluster);
+  k_dim.y = getDeviceAttr(cnrtAttrClusterCount);
+  k_dim.z = 1;
+  cnrtDataType_t data_type = getCnrtDataType(input.elemType());
+
+  KernelRoiAlign(k_dim, k_type, queue, data_type, (void*)input_ptr->data(),
+                 rois.data(), channels, aligned, aligned_height, aligned_width,
+                 height, width, sampling_ratio, spatial_scale, num_rois,
+                 (void*)output_ptr->data());
+#if 1
```
Comment (zhaoguochun1995) [inline mmcv/ops/csrc/parrots/mlu/roi_align_mlu.cpp:86]: 调试代码需要删掉

File: setup.py [hunk 3]
```diff
@@ -202,29 +202,54 @@ def get_extensions():
...
+            mlu_kernel_files = [
+                './mmcv/ops/csrc/common/mlu/roi_align_mlu_kernel.mlu',
+                './mmcv/ops/csrc/common/mlu/nms_mlu_kernel.mlu',
+                './mmcv/ops/csrc/common/mlu/focal_loss_sigmoid_mlu_kernel.mlu'
             ]
+            op_files = mlu_kernel_files +\
+                glob.glob('./mmcv/ops/csrc/parrots/common/*.cpp') +\
+                glob.glob('./mmcv/ops/csrc/parrots/cpu/*.cpp') +\
+                glob.glob('./mmcv/ops/csrc/parrots/mlu/*.cpp') +\
+                glob.glob('./mmcv/ops/csrc/pytorch/cpu/*.cpp')
```
Comment (chenyuwang814) [inline setup.py:245]: 这里是不是写错了

Comment (yewentao256) [reply setup.py:245]: cpu的kernel用到了pytorch的cpu kernel，check了一下没问题

File: mmcv/ops/csrc/parrots/mlu/roi_align_mlu.cpp (hunk 2 above)
Comment (yewentao256) [reply mmcv/ops/csrc/parrots/mlu/roi_align_mlu.cpp]: 已处理

**Comment by yewentao256** [comment]
close this issue since the functionality is no needed now

**Pull request closed without merging**